### PR TITLE
fix: message input

### PR DESCRIPTION
**Pull request by deven98**
added issue fixes

### CODE REVIEW

**Comment by xsahil03x** [comment]
@deven98 Let me share what @imtoori had in his mind about the builder

```dart
typedef MentionTileBuilder = Widget Function(BuildContext context, User user);

class MessageInput extends StatelessWidget {

  final MentionTileBuilder mentionTileBuilder;

  @override
  Widget build(BuildContext context) {
    return Container(
      child: mentionTileBuilder(user) ?? DefaultMentionTile(user: user),
    );
  }
}
```

```dart
class DefaultMentionTile extends StatelessWidget {
  final User user;
  final String title;
  final String subtitle;
  final Widget leading;
  final Widget trailing;
  
  @override
  Widget build(BuildContext context) {
    return ListTile(
      leading: leading ?? DefaultLeading(),
      trailing: trailing ?? DefaultTrailing(),
      title: title ?? DefaultTitle(),
      subtitle: subtitle ?? DefaultSubtitle(),
    );
  }
}
```

Then we can either return our own custom widget or do something like this.
```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MessageInput(
      mentionTileBuilder: (user) => DefaultMentionTile(
        user : user,
        subtitle: 'myCustomSubtitle',
      ),
    )
  }
}
```

**Comment by xsahil03x** [comment]
LGTM, just can we change the name from `DefaultMentionTile` to something `Stream Chat` specific as it will be easier to identify and use?

File: packages/stream_chat_flutter/lib/src/default_mention_tile.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+class DefaultMentionTile extends StatelessWidget {
+  final Member member;
+  final Widget title;
+  final Widget subtitle;
+  final Widget leading;
+  final Widget trailing;
+
+  DefaultMentionTile(this.member,
+      {this.title, this.subtitle, this.leading, this.trailing});
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/default_mention_tile.dart:13]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/default_mention_tile.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                        '@${member.userId}',
+                        maxLines: 1,
+                        overflow: TextOverflow.ellipsis,
+                        style: StreamChatTheme.of(context)
+                            .textTheme
+                            .footnoteBold
+                            .copyWith(
+                                color: StreamChatTheme.of(context)
+                                    .colorTheme
+                                    .grey),
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/default_mention_tile.dart:66]: missing trailing comma

File: packages/stream_chat_flutter/lib/src/message_input.dart [hunk 3]
```diff
@@ -32,6 +31,9 @@ typedef AttachmentThumbnailBuilder = Widget Function(
   Attachment,
 );
 
+typedef MentionTileBuilder = Widget Function(
+    BuildContext context, Member member);
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_input.dart:35]: missing trailing comma

**Review by imtoori — CHANGES_REQUESTED** [review]
A couple of trailing commas are missing
I'd change the name from `DefaultMentionTile` to just `MentionTile`, what do you think? (cc @xsahil03x)
Otherwise LGTM

**Comment by xsahil03x** [comment]
@imtoori yes looks fine to me 👍🏼

File: packages/stream_chat_flutter/lib/src/default_mention_tile.dart [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import 'package:flutter/material.dart';
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/default_mention_tile.dart:1]: missing docs in the whole class

File: packages/stream_chat_flutter/lib/src/message_input.dart [hunk 5]
```diff
@@ -32,6 +31,11 @@ typedef AttachmentThumbnailBuilder = Widget Function(
   Attachment,
 );
 
+typedef MentionTileBuilder = Widget Function(
```
Comment (imtoori) [inline packages/stream_chat_flutter/lib/src/message_input.dart:34]: missing docs

**Review by imtoori — CHANGES_REQUESTED** [review]
missing docs and also, can you rename the default_mention_tile.dart file to mention_tile.dart?

**Review by imtoori — APPROVED** [review]
LGTM

**Pull request merged**